File: discovery-server/src/main/java/app/metatron/discovery/domain/dataprep/entity/Dataset.java
package app.metatron.discovery.domain.dataprep.entity;

import app.metatron.discovery.domain.AbstractHistoryEntity;
import app.metatron.dataprep.teddy.DataFrame;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonValue;
import org.hibernate.annotations.GenericGenerator;
import javax.persistence.*;
import javax.validation.constraints.Size;

@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
@Entity
@Table(name = "dataset")
public class Dataset extends AbstractHistoryEntity {

    @JsonFormat(shape = JsonFormat.Shape.OBJECT)
    public enum IMPORT_TYPE {
        UPLOAD,
        URI,
        DATABASE,
        KAFKA;
        @JsonValue
        public String toJson() {
            return name();
        }
    }

    @JsonFormat(shape = JsonFormat.Shape.OBJECT)
    public enum FILE_FORMAT {
        CSV,
        EXCEL,
        JSON;

        @JsonValue
        public String toJson() {
            return name();
        }
    }

    @JsonFormat(shape = JsonFormat.Shape.OBJECT)
    public enum RS_TYPE {
        TABLE,
        QUERY;

        @JsonValue
        public String toJson() {
            return name();
        }
    }

    @Id
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid2")
    @Column(name = "ds_id")
    String dsId;

    @Size(max = 2000)
    @Column(name = "name", nullable = false)
    private String name;

    @Lob
    @Column(name = "description")
    private String description;

    @Lob
    @Column(name = "custom")
    private String custom;

    @Column(name = "import_type")
    @Enumerated(EnumType.STRING)
    private Dataset.IMPORT_TYPE importType;

    @Column(name = "conn_id")
    private String connId;

    @Column(name = "rs_type")
    @Enumerated(EnumType.STRING)
    private Dataset.RS_TYPE rsType;


    @Size(max = 255)
    @Column(name = "db_name")
    private String dbName;

    @Size(max = 255)
    @Column(name = "tbl_name")
    private String tblName;

    @Lob
    @Column(name = "query_stmt")
    private String queryStmt;

    @Column(name = "file_format")
    @Enumerated(EnumType.STRING)
    private Dataset.FILE_FORMAT fileFormat;

    @Lob
    @Column(name = "filename_before_upload")
    private String filenameBeforeUpload;

    @Size(max = 2000)
    @Column(name = "sheet_name")
    private String sheetName;

    @Lob
    @Column(name = "stored_uri")
    private String storedUri;

    @Column(name = "delimiter")
    private String delimiter;

    @Column(name = "quote_char")
    private String quoteChar;

    @Lob
    @Column(name = "serialized_preview")
    String serializedPreview;

    @Column(name = "manual_column_count")
    private Integer manualColumnCount;

    @Column(name = "total_lines")
    private Long totalLines;

    @Column(name = "total_bytes")
    private Long totalBytes;

    @Transient
    @JsonProperty
    DataFrame gridResponse;


    public String getDsId() {
        return dsId;
    }

    public void setDsId(String dsId) {
        this.dsId = dsId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getCustom() {
        return custom;
    }

    public void setCustom(String custom) {
        this.custom = custom;
    }

    public IMPORT_TYPE getImportType() {
        return importType;
    }

    public void setImportType(IMPORT_TYPE importType) {
        this.importType = importType;
    }

    public String getConnId() {
        return connId;
    }

    public void setConnId(String connId) {
        this.connId = connId;
    }

    public RS_TYPE getRsType() {
        return rsType;
    }

    public void setRsType(RS_TYPE rsType) {
        this.rsType = rsType;
    }

    public String getDbName() {
        return dbName;
    }

    public void setDbName(String dbName) {
        this.dbName = dbName;
    }

    public String getTblName() {
        return tblName;
    }

    public void setTblName(String tblName) {
        this.tblName = tblName;
    }

    public String getQueryStmt() {
        return queryStmt;
    }

    public void setQueryStmt(String queryStmt) {
        this.queryStmt = queryStmt;
    }

    public FILE_FORMAT getFileFormat() {
        return fileFormat;
    }

    public void setFileFormat(FILE_FORMAT fileFormat) {
        this.fileFormat = fileFormat;
    }

    public String getFilenameBeforeUpload() {
        return filenameBeforeUpload;
    }

    public void setFilenameBeforeUpload(String filenameBeforeUpload) {
        this.filenameBeforeUpload = filenameBeforeUpload;
    }

    public String getSheetName() {
        return sheetName;
    }

    public void setSheetName(String sheetName) {
        this.sheetName = sheetName;
    }

    public String getStoredUri() {
        return storedUri;
    }

    public void setStoredUri(String storedUri) {
        this.storedUri = storedUri;
    }

    public String getDelimiter() {
        return delimiter;
    }

    public void setDelimiter(String delimiter) {
        this.delimiter = delimiter;
    }

    public String getQuoteChar() {
        return quoteChar;
    }

    public void setQuoteChar(String quoteChar) {
        this.quoteChar = quoteChar;
    }

    public String getSerializedPreview() {
        return serializedPreview;
    }

    public void setSerializedPreview(String serializedPreview) {
        this.serializedPreview = serializedPreview;
    }

    public Integer getManualColumnCount() {
        return manualColumnCount;
    }

    public void setManualColumnCount(Integer manualColumnCount) {
        this.manualColumnCount = manualColumnCount;
    }

    public Long getTotalLines() {
        return totalLines;
    }

    public void setTotalLines(Long totalLines) {
        this.totalLines = totalLines;
    }

    public Long getTotalBytes() {
        return totalBytes;
    }

    public void setTotalBytes(Long totalBytes) {
        this.totalBytes = totalBytes;
    }

    public DataFrame getGridResponse() {
        return gridResponse;
    }

    public void setGridResponse(DataFrame gridResponse) {
        this.gridResponse = gridResponse;
    }
}

File: prep-frontend/src/app/dataset/domains/dataset.ts
/* tslint:disable */
import {AbstractHistory} from '../../common/domain/abstract-history-entity';
import {PageResult} from '../../common/constants/page';
import {Connection} from '../../connection/domains/connection';

export namespace Dataset {

  export class Entity extends AbstractHistory.Entity {
    dsId: string;
    name: string;
    description: string;
    custom: string;
    importType: IMPORT_TYPE;
    connId: string;
    rsType: RS_TYPE;
    dbName: string;
    tblName: string;
    queryStmt: string;
    fileFormat: FILE_FORMAT;
    filenameBeforeUpload: string;
    sheetName: string;
    storedUri: string;
    delimiter: string;
    quoteChar: string;
    serializedPreview: string;
    manualColumnCount: number;
    totalLines: number;
    totalBytes: number;
  }

  export class Select extends Entity {
    implementor: string;
    hostname: string;
    port: string;
    database: string;
    catalog: string;
    sid: string;
    url: string;
    connType: Connection.ConnType;
    gridResponse: any;

  }

  export class DatasetFile extends Entity {
    public sheets: string[];
    public sheetIndex: number;
    public selectedSheets: string[] = [];
    public sheetInfo: SheetInfo[];
    public fileName: string;
    public fileExtension: string;
    public selected = false;
    public error: object;
    public storageType: StorageType;
  }

  export class DatasetDatabase extends Entity {
    public connectionName: string;
    public implementor: string;

  }


  export class SheetInfo {
    selected: boolean;
    data: object[];
    fields: object[] = [];
    totalRows?: number;
    valid: boolean;
    sheetName?: string;
    columnCount?: number;
  }

  export class SimpleListEntity extends AbstractHistory.Entity {
    selected: boolean;
    origin: boolean;
    dsId: string;
    name: string;
    description: string;
    custom: string;
    importType: IMPORT_TYPE;
    rsType: RS_TYPE;
    dbName: string;
    tblName: string;
    queryStmt: string;
    fileFormat: FILE_FORMAT;
    filenameBeforeUpload: string;
    sheetName: string;
    storedUri: string;
    delimiter: string;
    quoteChar: string;
    serializedPreview: string;
    manualColumnCount: number;
    totalLines: number;
    totalBytes: number;
  }



  export enum IMPORT_TYPE {
    UPLOAD = 'UPLOAD',
    URI = 'URI',
    DATABASE = 'DATABASE',
    KAFKA = 'KAFKA'
  }

  export enum FILE_FORMAT {
    CSV = 'CSV',
    EXCEL = 'EXCEL',
    JSON = 'JSON',
    TXT = 'TXT'
  }

  export enum RS_TYPE {
    TABLE = 'TABLE',
    QUERY = 'QUERY'
  }
  export enum StorageType {
    LOCAL = 'LOCAL',
    HDFS = 'HDFS',
    S3 = 'S3',
    BLOB = 'BLOB',
    FTP = 'FTP',
  }




  export namespace Result {
    export namespace GetDatasets {
      export class SearchedData {
        page: PageResult;
        '_embedded': {
          datasets: Array<Dataset.SimpleListEntity>
        };
      }
    }
  }

}
